### PR TITLE
Add context and info to logging interface

### PR DESCRIPTION
**Pull request by jaronoff97**
Allow the logging interface to propagate context where possible. This will let logging sinks contain richer information about potential opamp IO errors.

### CODE REVIEW

**Comment by jaronoff97** [comment]
Before i continue with this... @andykellr @evan-bradley @tigrannajaryan what do you all think of this approach? Propagating the context would make it easier for consumers to understand the potential problems that the impls run in to, but i'm not sure how i feel about the interface. Let me know your thoughts. Thank you.

**Comment by tigrannajaryan** [comment]
I have a hard time reviewing the code due to all the junk codecov added. I am going to disable it for now.

**Comment by tigrannajaryan** [comment]
The idea to pass the context to logging functions sounds good to me. I am not sure we need a separate set of functions for that. Perhaps just add context parameter to existing functions?

**Comment by jaronoff97** [comment]
@tigrannajaryan sounds good! I'll update to do that.

**Comment by mwear** [comment]
One benefit to the current approach is that the interface matches the new Go structured logger, [slog](https://pkg.go.dev/log/slog#Logger). Rather than adding context to the existing functions, it might make more sense to keep the context versions, and remove (or deprecate) the non-context versions. That way you can use any logger that is `slog` compatible.

**Comment by jaronoff97** [comment]
yeah that was the intended purpose... I see benefits for both approaches... Maybe we should discuss at the next SIG call?

**Comment by tigrannajaryan** [comment]
The `Logger` interface here is not a general-purpose logging API. It is on purpose minimalistic and designed to support only what is needed by opamp implementation and nothing else. If we want to pass the context I am happy to make it a required parameter, I don't see the need to give a choice (side note: I argued that slog also makes it a required parameter, but wasn't able to convince the author).

I prefer to keep Logger interface as small as possible and bridge to any other logging API as needed. Coding a bridge from `Logger` to `slog` is so easy that I don't see the need to make the interface identical.

We also do not have stability promises in this repo yet, so I am not too worried about making a breaking change that is very easy to adopt. I don't think we need a deprecations process for this.

**Comment by jaronoff97** [comment]
done! I will also work on the follow up described [here](#214) which will make these even more valuable.

File: client/types/logger.go [hunk 1]
```diff
@@ -1,7 +1,10 @@
 package types
 
+import "context"
+
 // Logger is the logging interface used by the OpAMP Client.
 type Logger interface {
-	Debugf(format string, v ...interface{})
-	Errorf(format string, v ...interface{})
+	Infof(ctx context.Context, format string, v ...interface{})
```
Comment (tigrannajaryan) [inline client/types/logger.go:7]: This does not seem to be used anywhere. Can we remove?

**Review by tigrannajaryan — COMMENTED** [review]
Mostly LGTM, except I think we don't need the Infof() for now. Can be added later when needed.

**Comment by jaronoff97** [comment]
the codecov is failing for probably obvious reasons – testing error logs isn't really done right now

**Review by tigrannajaryan — APPROVED** [review]

**Comment by tigrannajaryan** [comment]
This is a breaking change in the API, allowed because the API is not declared stable yet.

**Pull request merged**